applications/ShallowWaterApplication/custom_processes/id_renumbering_process.h

```diff
@@ -0,0 +1,259 @@
+
+#ifndef KRATOS_ID_RENUMBERING_PROCESS_H_INCLUDED
+#define KRATOS_ID_RENUMBERING_PROCESS_H_INCLUDED
+
+
+// System includes
+#include <unordered_map>
+
+
+// External includes
+
+
+// Project includes
+#include "processes/process.h"
+#include "containers/model.h"
+
+
+namespace Kratos
+{
+///@addtogroup ShallowWaterApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+typedef std::size_t IndexType;
+typedef std::vector<std::string> StringVectorType;
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This utility renumber all the nodes, elements and/or conditions inside a model.
+/** Gid needs a unique Id for all the entities. Duplicated Ids across several model
+ *  parts won't be printed correctly.
+*/
+class KRATOS_API(SHALLOW_WATER_APPLICATION) IdRenumberingProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    typedef Node<3> NodeType;
+
+    /// Pointer definition of IdRenumberingProcess
+    KRATOS_CLASS_POINTER_DEFINITION(IdRenumberingProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    IdRenumberingProcess(Model& rThisModel);
+
+    IdRenumberingProcess(Model& rThisModel, const StringVectorType& rModelPartNames);
+
+    /// Destructor.
+    virtual ~IdRenumberingProcess() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void RenumberNodes();
+
+    void RenumberElements();
+
+    void RenumberConditions();
+
+    void RestoreNodes();
+
+    void RestoreElements();
+
+    void RestoreConditions();
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const override
+    {
+        return "IdRenumberingProcess";
+    }
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const override {}
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    Model& mrModel;
+    StringVectorType mModelPartNames;
+    std::unordered_map<IndexType, IndexType> mOriginNodesIds;
```

Comment:
I'd rename these member variables to something like `mOriginNodesIdsMap`(until I arrived to this point I didn't realize it is a map).

Reply:
Renamed